### ISSUE TITLE
Add prefer-fp

### ISSUE THREAD

**Issue by epeli** [wix/eslint-plugin-lodash#35]
I'm using `lodash/fp` exclusively. It would be nice if it where possible to force consistent usage of the fp version (and maybe vice versa for the vanilla version).

So these would be warnings:

```js
import _ from "lodash";
import mapValues from "lodash/mapValues";
```

and these not

```js
import _ from "lodash/fp";
import mapValues from "lodash/fp/mapValues";
```

**Comment by ganimomer** [comment]
Hi,
This sounds like a nice idea, we can implement it (or accept PRs, of course :smile:)

As a side note, you should know that  we don't support `lodash/fp` in any of our other rules, and it's not in the works at the moment. (The rules make assumptions that don't apply there, like the location of the `iteratee` parameter and functions not being curried, and the plugin generally favours `_.chain` over `_.flow` and `_.flowRight`, which are the more functional way).

**Comment by camjc** [comment]
It would also be great to be able to warn against mutating methods (defaults preferred over assign etc).

**Comment by jfmengels** [comment]
I have just started working on https://github.com/jfmengels/eslint-plugin-lodash-fp, where I implemented this rule.

I started working on it because I saw @ganimomer's comment on the fact that `lodash/fp` is not supported, and far from it (a lot of rework would be needed as I see it).

Rules specific to `lodash-fp` would probably be a bit simpler to implement too, as you don't have `_.chain()` (or at least it's discouraged, the second rule I implemented is forbidding chaining).

Thing is, a lot of rules would make sense in both of them (`path-style`, `prefer-compact`, etc...) though the implementation in this repo will not work because they rely on the param order.
I will probably want to copy them to lodash-fp, but wanted to let you guys know at least, and know if you had any problems with that.

**Comment by ganimomer** [comment]
@jfmengels, Thanks for the heads up! Looking forward to see how it turns out.
`lodash/fp` is great, but since we don't use it much in our own codebase and it would require a **lot** of rework, we just aren't getting to it. 
It's awesome that this need is getting answered, and I personally intend to contribute.

That settles it then! People who want to use `lodash/fp` are encouraged to `npm install eslint-plugin-lodash-fp`.
